### PR TITLE
feat(websocket): Added new API `esp_websocket_client_append_header` (IDF-7893)

### PR DESCRIPTION
**Pull request by gabsuren**
Added new API esp_websocket_client_append_header
Also updated host example, to enhance covertly analyzing

### CODE REVIEW

File: components/esp_websocket_client/include/esp_websocket_client.h [hunk 1]
```diff
@@ -155,16 +155,36 @@ esp_websocket_client_handle_t esp_websocket_client_init(const esp_websocket_clie
...
+ * @brief      Sets additional WebSocket headers for the client
+ * @pre        The WebSocket client must be stopped before setting headers if it is already connected.
  *
- * @param[in]  client  The client
- * @param headers  additional header strings each terminated with \r\n
+ *  Notes:
+ *  - When setting headers using this method, any previous headers will be replaced.
+ *    To set multiple headers, you can concatenate them using the "\r\n" delimiter.
+ *    Alternatively, use the `esp_websocket_client_append_header` function for individual header additions.
+ *    Example: websocket_cfg.headers = "Sec-WebSocket-Key: my_key\r\nPassword: my_pass\r\n";
```
Comment (david-cermak) [inline components/esp_websocket_client/include/esp_websocket_client.h:165]: I'd swap these two lines
```suggestion
 *    Example: websocket_cfg.headers = "Sec-WebSocket-Key: my_key\r\nPassword: my_pass\r\n";
 *    Alternatively, use the `esp_websocket_client_append_header` function for individual header additions.
```

File: components/esp_websocket_client/examples/linux/main/main.c [hunk 2]
```diff
@@ -79,6 +79,9 @@ static void websocket_app_start(void)
     ESP_LOGI(TAG, "Connecting to %s...", websocket_cfg.uri);
 
     esp_websocket_client_handle_t client = esp_websocket_client_init(&websocket_cfg);
+    // This call has no operational effect; it's used for testing and enhancing coverity analysis
```
Comment (david-cermak) [inline components/esp_websocket_client/examples/linux/main/main.c:82]: This is still an example, so we should say that this code demonstrates adding another header

Comment (david-cermak) [reply components/esp_websocket_client/examples/linux/main/main.c:82]: Also, it doesn't *enhance* coverity analysis. It improves code coverage.
```suggestion
.... (called to increase code coverage)
```

(note that Coverity is a tool and this doesn't impact any analysis, but the result/metrics)

File: components/esp_websocket_client/include/esp_websocket_client.h [hunk 3]
```diff
@@ -155,16 +155,36 @@ esp_websocket_client_handle_t esp_websocket_client_init(const esp_websocket_clie
...
- * @param[in]  client  The client
- * @param headers  additional header strings each terminated with \r\n
+ *  Notes:
+ *  - When setting headers using this method, any previous headers will be replaced.
+ *    To set multiple headers, you can concatenate them using the "\r\n" delimiter.
+ *    Example: websocket_cfg.headers = "Sec-WebSocket-Key: my_key\r\nPassword: my_pass\r\n";
+ *    Alternatively, use the `esp_websocket_client_append_header` function for individual header additions.
+ *
+ * @param[in]  client  The WebSocket client handle.
+ * @param      headers Additional header strings, each terminated with \r\n.
```
Comment (david-cermak) [inline components/esp_websocket_client/include/esp_websocket_client.h:168]:
```suggestion
 * @param[in]      headers Additional header strings, each terminated with \r\n.
```

Comment (david-cermak) [reply components/esp_websocket_client/include/esp_websocket_client.h:168]: Question: Are these header strings really **"Additional"** ?

File: components/esp_websocket_client/include/esp_websocket_client.h [hunk 4]
```diff
@@ -155,16 +155,36 @@ esp_websocket_client_handle_t esp_websocket_client_init(const esp_websocket_clie
...
- * @pre        Must stop the WebSocket client before set headers if the client has been connected
+ * @brief      Sets additional WebSocket headers for the client
+ * @pre        The WebSocket client must be stopped before setting headers if it is already connected.
  *
- * @param[in]  client  The client
- * @param headers  additional header strings each terminated with \r\n
+ *  Notes:
+ *  - When setting headers using this method, any previous headers will be replaced.
+ *    To set multiple headers, you can concatenate them using the "\r\n" delimiter.
+ *    Example: websocket_cfg.headers = "Sec-WebSocket-Key: my_key\r\nPassword: my_pass\r\n";
```
Comment (david-cermak) [inline components/esp_websocket_client/include/esp_websocket_client.h:164]: Also this is an example of setting the headers as a part of config option. Should we clearly say what are the options that users have? For example, say you can:
1) Set headers directly in the config options (each item terminated by [CR][LF])
2) Use this API, (which will overwrite (1), ... )
3) Use the *append* API, which adds a single header

**Review by david-cermak — COMMENTED** [review]
LGTM otherwise

File: components/esp_websocket_client/include/esp_websocket_client.h (hunk 3 above)
Comment (gabsuren) [reply components/esp_websocket_client/include/esp_websocket_client.h:168]: During initial handshake, websocket client sends HTTP headers
```
I (6069) transport_ws: Write upgrade request
GET / HTTP/1.1
Connection: Upgrade
Host: 192.168.0.197:8080
User-Agent: ESP32 Websocket Client
Upgrade: websocket
Sec-WebSocket-Version: 13
Sec-WebSocket-Key: eW5wVnWNtkRMyw9KZwZDeg==
```
And those keys which we set in `esp_websocket_client_append_header `  appended to those headers. Like
`esp_websocket_client_append_header(client, "Password", "my_password")` the result will be duting handshake
```
I (6069) transport_ws: Write upgrade request
GET / HTTP/1.1
Connection: Upgrade
Host: 192.168.0.197:8080
User-Agent: ESP32 Websocket Client
Upgrade: websocket
Sec-WebSocket-Version: 13
Sec-WebSocket-Key: eW5wVnWNtkRMyw9KZwZDeg==
Password: my_password
```

Comment (david-cermak) [reply components/esp_websocket_client/include/esp_websocket_client.h]: Ok, got it. Thanks for explaining! The *additional(s)*  in the docs makes sense then.

File: components/esp_websocket_client/examples/target/main/websocket_example.c [hunk 5]
```diff
@@ -146,6 +146,34 @@ static void websocket_app_start(void)
...
+    esp_websocket_client_append_header(client, "aaa", "bbb");
+    esp_websocket_client_append_header(client, "aaa", "bbb");
+    esp_websocket_client_append_header(client, "aaa", "bbb");
+    esp_websocket_client_append_header(client, "aaa", "bbb");
+    esp_websocket_client_append_header(client, "aaa", "bbb");
+    esp_websocket_client_append_header(client, "aaa", "bbb");
+    esp_websocket_client_append_header(client, "aaa", "bbb");
+    esp_websocket_client_append_header(client, "aaa", "bbb");
+    esp_websocket_client_append_header(client, "aaa", "bbb");
+    esp_websocket_client_append_header(client, "aaa", "bbb");
```
Comment (david-cermak) [inline components/esp_websocket_client/examples/target/main/websocket_example.c:175]: Is this intentional, or did you just want to test something?

Comment (gabsuren) [reply components/esp_websocket_client/examples/target/main/websocket_example.c:175]: Thanks for catching that; it was unintentional :D

**Review by david-cermak — APPROVED** [review]

**Pull request merged**